### ISSUE TITLE
Application not removed from start menu once you unregister ubuntu distro in WSL2 G

### ISSUE THREAD

**Issue by anirugu** [microsoft/WSL#6893]
### Windows Build Number

21364

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Distro Version

20.04

### Repro Steps

I download the appx file but it's never work for me, Install the appx and reset the app nothing worked. So I read on github that you can extract it and install it by running exe. 

since I am on insider I tried wsl --install which install some stuff and later I install the distro and it's worked. 

Later I unregister the distro and deleted the ubuntu folder that I extracted from appx. I can still see the links in start menu which is not working because distro is gone.

I installed the ubuntu again by extract again and run ubuntu. First it's give me some error I unregister from cmd to make it work. I can use the ubuntu again but those apps links in start menu are not gone. 

### Expected Behavior

Expected behavior : If I unregistered a linux distro then the apps it's added into start menu should be gone as same time when I unregister. this make no sense that links is still there which will not work again.

### Actual Behavior

links not removed.

**Comment by anirugu** [comment]
It might not be a issue, I install the WSLG again and now I can see only transmission which I install in new distro.

**Comment by therealkenc** [comment]
> If I unregistered a linux distro then the apps it's added into start menu should be gone

It is nonobvious behavior, which could maybe be [documented](https://github.com/MicrosoftDocs/WSL/issues) better. The `wsl.exe --unregister` command leaves the start menu (aka distribution app) alone by-design. After unregistering, starting the distribution app from the start menu performs a fresh distribution install. Or, you could call it, "re-registers" the distribution. To uninstall the app itself, right click and uninstall from the start menu.

> and deleted the ubuntu folder that I extracted from appx

If you deleted a path manually out from under the start menu, that is going to result in the menu pointing nowhere. This behavior is not unique to WSL. The start links don't know it is gone.

> I download the appx file but it's never work for me

Please start a new issue submission for "never worked" under a different cover with as much detail as possible to repro if you like, unless you are unblocked sufficiently with `wsl.exe --install`. That is a better method for installing distributions with recent WSL regardless of what might have happened there.
